internal/provider/resource_tfe_workspace_run_task.go

```diff
@@ -297,3 +320,84 @@ func (r *resourceWorkspaceRunTask) UpgradeState(ctx context.Context) map[int64]r
 		},
 	}
 }
+
+func (r *resourceWorkspaceRunTask) supportsStagesProperty() bool {
+	// The Stages property is available in HCP Terraform and Terraform Enterprise v202404-1 onwards.
+	//
+	// The version comparison here can use plain string comparisons due to the nature of the naming scheme. If
+	// TFE every changes its scheme, the comparison will be problematic.
+	return r.capabilities.IsCloud() || r.capabilities.RemoteTFEVersion() > "v202404"
+}
+
+func (r *resourceWorkspaceRunTask) addStageSupportDiag(d *diag.Diagnostics, isError bool) {
+	summary := "Terraform Enterprise version"
+	detail := fmt.Sprintf("The version of Terraform Enterprise does not support the stages attribute on Workspace Run Tasks. Got %s but requires v202404-1+", r.config.Client.RemoteTFEVersion())
+	if isError {
+		d.AddError(detail, summary)
+	} else {
+		d.AddWarning(detail, summary)
+	}
+}
+
+func (r *resourceWorkspaceRunTask) extractStageAndStages(plan modelTFEWorkspaceRunTaskV1, d *diag.Diagnostics) (*tfe.Stage, *[]tfe.Stage) {
```

Comment:
Are we certain we want a pointer to a slice? A slice can have the value `nil` if you were wanting to validate if a value was returned.

Reply:
Whoops, yeah I forgot about that... I can change that

Reply:
Done